web/src/test/java/org/cbioportal/weblegacy/CNASegmentControllerTest.java

```diff
@@ -0,0 +1,118 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+package org.cbioportal.weblegacy;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.cbioportal.model.CNASegmentData;
+import org.cbioportal.service.CNASegmentService;
+import org.cbioportal.web.config.CustomObjectMapper;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@WebAppConfiguration
+@ContextConfiguration(classes = {CNASegmentControllerConfig.class, CustomObjectMapper.class})
+public class CNASegmentControllerTest {
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+    @Autowired
+    private CNASegmentService cNASegmentServiceMock;
+    private MockMvc mockMvc;
+
+    @Before
+    public void setup() {
+        Mockito.reset(cNASegmentServiceMock);
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+    }
+    
+    @Test
+    public void getCNASegmentTest() throws Exception {
+        List<CNASegmentData> mockResponse = new ArrayList<>();
+        CNASegmentData cNASegmentData1 = new CNASegmentData(); 
+        cNASegmentData1.setStart(148350479);
+        cNASegmentData1.setEnd(158385118);
+        cNASegmentData1.setChr("7");
+        cNASegmentData1.setNumProbes(4901);
+        cNASegmentData1.setValue((float) 0.1315);
+        cNASegmentData1.setSample("TCGA-AG-3732-01");
+        
+        CNASegmentData cNASegmentData2 = new CNASegmentData(); 
+        cNASegmentData2.setStart(148347132);
+        cNASegmentData2.setEnd(148348416);
+        cNASegmentData2.setChr("7");
+        cNASegmentData2.setNumProbes(3);
+        cNASegmentData2.setValue((float) -1.5009);
+        cNASegmentData2.setSample("TCGA-AG-3732-01");
+       
```

Comment:
two blank lines here is excessive --- consider having no blank line here or between other statements within this function.